### ISSUE TITLE
Cannot send messages using Microsoft Bot Emulator

### ISSUE THREAD

**Issue by Saikat786** [Microsoft/BotFramework-Emulator#226]
Version 3.5.29

The emulator at first is supposed to return a response with the number of characters of a user input string. However even after using the default endpoint url which is http://localhost:3979/api/messages, whenever inputting a string, the message is not sent.

**What appears in the log is:**
`[04:31:00] Emulator listening on http://[::]:55944 
[04:31:01] Failed to start ngrok: undefined 
[04:31:01] ngrok may already be running in a different process. ngrok's free tier allows only one instance at a time per host. 
[04:31:01] Checking for new version... 
[04:31:02] Application is up to date.`

**Comment by eanders-MS** [comment]
When you try to send your bot a message, does it generate an entry in the emulator's log?

**Comment by eanders-MS** [comment]
Closing due to inactivity. Reply here if you still need assistance and I will reopen.

**Comment by eanders-MS** [comment]
@fr6087 Can you tell me about your setup:
- Are you running your bot locally (on the same machine as the emulator)?
- Are you running with or without credentials?
- Are any errors showing up in your emulator log?

Following the steps outlined in the Authentication Troubleshooting Guide is a great way to ensure your configuration is setup correctly. https://docs.microsoft.com/en-us/bot-framework/troubleshoot-authentication-problems

**Comment by fr6087** [comment]
Yes, I'm running under http://localhost:3979/api/messages without security enabled that is I checked that there are no values in the Webconfig-file for appsettings-credentials  
<add key="MicrosoftAppId" value="" />
    <add key="MicrosoftAppPassword" value="" />
  As I said before, the error I get initially is ngrok undefined (though I downloaded ngrok separate and pointed to the .exe in App Settings) I read that it is not needed for local Debugging so I checked "disable for local addresses".  
Let me show you a picture...
![image](https://user-images.githubusercontent.com/31841576/31010260-8c92814e-a50a-11e7-99ac-5caf8945ac58.png)

**Comment by Tushti02** [comment]
[14:15:13] Checking for new version...
 [14:15:14] Failed to start ngrok: undefined
 [14:15:14] ngrok may already be running in a different process. ngrok's free tier allows only one instance at a time per host.

Hi,
 I am also facing this issue while using Bot framework channel Emulator and ngrok tunneling software.
 I gave the proper path in App settings of emulator for ngrok.exe and had also launched emulator as administrator. But end up in above error.
 If anyone can help in this.

Thanks !

**Comment by fr6087** [comment]
I found out that the emulator was connecting to the **proxy** of my company instead of dissolving localhost. I don't know all the steps because a collegue was helping me with this. But we used Postman (that's a Google Chrome Extension) for monitoring POST/GET traffic and then modified proxy settings in Postman. It resulted in a workaround with copy-pasting JSON from Botframework Emulator zu Postman.  Workaround worked, Bot answered. But in the emulator itself, I saw no possibility to adapt proxy settings and Botframework Emulator is still not working whenever I am using the company Network.
